### DISCUSSION TITLE
haskell-template now has dedicated documentation

### DISCUSSION THREAD

**Discussion by srid** [srid/haskell-template#60] [Announcements]
https://srid.ca/haskell-template

I hope to document "how to do X" kind of things here.

**Comment by billksun** [comment]
I guess as a first "how to" question from a beginner: 
How do I specify a GHC version to use?

**Reply by srid** [reply]
Good idea; noted in https://github.com/srid/haskell-template/issues/41

**Comment by subterfugue** [comment]
How would someone add a dependency in the flake, non-haskell related?

**Reply by srid** [reply]
Example: emanote adds `stork` as both [build-time](https://github.com/srid/emanote/blob/66dad06ca8fff73e30c52aee28f6458c73c900e3/flake.nix#L42-L43) (comes from [here](https://github.com/srid/emanote/blob/master/nix/stork.nix)) and [runtime](https://github.com/srid/emanote/blob/66dad06ca8fff73e30c52aee28f6458c73c900e3/flake.nix#L78) dependency.

**Reply by subterfugue** [reply]
Interesting, that project has weird ways to define hls and ghcid, they're done differently in this generic template.

**Reply by srid** [reply]
Are you referring to the use of attrsets instead of lists? The `buildTools` option come from https://github.com/srid/haskell-flake and it was @roberth 's suggestion originally to go with attrsets (because you can then override the defaults on a per-dependency basis; as Emanote demonstrates with ghcid, etc. all). TBH, I don't know of better ways to do it.

**Comment by srid** [comment]
Also: https://haskell.flake.page/

**Comment by subterfugue** [comment]
I find the adding a package from Nixpkgs into the flake something that I do frequently and there should be an option to do this easily (and especially documentation for this, which is why i posted this here... add a "how to add an input" option?), because otherwise it seems rather confusing and error-prone to me. I have https://termpad.com/UglyDifficultZebra
which is erroring because
```
error: attribute 'llvm' missing

       at /nix/store/4w4df1qqjxy58xyzxszk47gnlmdpg5d2-source/flake.nix:33:26:

           32|           devShell = {
           33|             tools = hp: {
             |                          ^
           34|               treefmt = config.treefmt.build.wrapper;
(use '--show-trace' to show detailed location information)

```
I tried to use `inherit (config.packages.llvmPackages_15) llvm;` first because that's the version i *actually* want, but as a last resort tried the above and it failed. I'm not quite sure what I'm doing wrong.

**Reply by srid** [reply]
@ribosomerocker Why are you using `config.packages`?

**Reply by subterfugue** [reply]
Because that's what I've see being used in the links you posted 😅

**Reply by srid** [reply]
```
inherit (config.packages) llvm;
```

This should be:

```
inherit (pkgs) llvm;
```

`config.packages` refers to the packages in your local flake which obviously doesn't export llvm.
